CMakeLists.txt

```diff
@@ -0,0 +1,342 @@
+cmake_minimum_required(VERSION 3.28)
+
+cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0091 NEW)
+
+set(PKG_VERSION 1.0.31)
+set(LIB_VERSION 1.74.0)
+
+project(tkrzw
+  VERSION ${PKG_VERSION}
+  DESCRIPTION "a set of implementations of DBM"
+  HOMEPAGE_URL "https://dbmx.net/tkrzw/"
+  LANGUAGES C CXX
+)
+add_library(tkrzw)
+set_target_properties(tkrzw PROPERTIES VERSION ${LIB_VERSION} SOVERSION 1)
+
+include(CMakePackageConfigHelpers)
+include(CPack)
+include(GNUInstallDirs)
+include(CMakeDependentOption)
+
+option(WITH_TOOLS "Build cli frontends" OFF)
+cmake_dependent_option(WITH_TESTS "Generate a test target" OFF "WITH_TOOLS" OFF)
+
+
+find_package(Threads)
+target_link_libraries(tkrzw PUBLIC Threads::Threads)
+
+# See https://gitlab.kitware.com/cmake/cmake/-/issues/18837
+if(MSVC)
+	target_compile_options(tkrzw PUBLIC "/Zc:__cplusplus")
+endif()
+
+target_compile_definitions(tkrzw
+    PRIVATE
+        _FILE_OFFSET_BITS=64
+        _TKRZW_PKG_VERSION="${CMAKE_PROJECT_VERSION}"
+        _TKRZW_LIB_VERSION="${LIB_VERSION}"
+)
+
+target_compile_features(tkrzw PUBLIC cxx_std_17)
```

Comment:
No opinion about this line, this means if the compiler by default uses 20 or 23 then the code will be compiled in 20 or 23 mode.

Alternatively, set the version to fixed 17

```
set_property(TARGET tkrzw PROPERTY CXX_STANDARD 17)
```